application/templates/secret.yaml

```diff
@@ -15,9 +15,21 @@ metadata:
   annotations:
 {{ toYaml $.Values.secret.annotations | indent 4 }}
 {{- end }}
+{{- if .data }}
 data:
 {{- range $key, $value := .data }}
   {{ $key }}: {{ $value | b64enc }}
 {{- end }}
+{{- else if .stringData }}
+stringData:
+{{- range $key, $value := .stringData }}
+  {{ $key }}: {{ $value }}
+{{- end }}
```

Comment:
I feel like these are now doing the exact same thing. While I'm all for options and choice, is there a real benefit to maintaining the status quo (encoding through Helm) when the Kubernetes API provides the write-only `stringData`? I also posit that perhaps `stringData` itself isn't necessary since it's just a way to delegate what we're already doing to the API server.

Reply:
Let me explain our requirement. We use 1pass vault plugin for Kubernetes, and the actual value is a placeholder that will be replaced by the plugin (it is an operator). Therefore, we need to set a string value, and leave it without being encoded at all, as the real value will be replaced by the plugin.

Reply:
Interesting! I'm not opposed to the change, but from what you're telling me I think you'd need an admission hook, no? When writing a `Secret` with `stringData` in it, the API server encodes the plain text into base 64 and then makes it a regular `data` key. According to the documentation, `stringData` is never output by the API.

Does this fit your scenario? I'm not sure if you could see `stringData` in an admission hook, but I bet that's what you might have to work with.

Reply:
i am not sure what you actually mean unfortunately. Could you explain me more? We are currently using the chart with this new feature on our k8s cluster from a local registry. I know it works the way we need because of it.

Reply:
My assumption when you mentioned the 1pass vault plugin behaved as an operator was that it would monitor Kubernetes resources as they were ingested into etcd and read from the API server. The documentation is clear that `stringData` for secrets are never obtained from the API server and exist solely as a write-only convenience method.

To be able to capture the field, therefore, one would have to create an admission hook, which runs before resources are sent to etcd to be read as internal state through the API server. However, since you mentioned your setup already works, I'll assume there's something I'm not seeing and the plugin works its way into the K8s API rather than behave as the kind of operator I'm used to!

For now I'm resolving this thread. You can disregard what I said!